framework/src/utils/MultiMooseEnum.C

```diff
@@ -235,57 +227,53 @@ void
 MultiMooseEnum::remove(InputIterator first, InputIterator last)
 {
   // Create a new list of enumerations by striping out the supplied values
-  std::set<std::string> current = _current_names;
   for (InputIterator it = first; it != last; ++it)
   {
     // Values stored as upper case
-    std::string upper(*it);
-    std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
-
-    std::set<std::string>::iterator found = current.find(upper);
-    if (found != current.end())
-      current.erase(found);
+    std::string upper(MooseUtils::toUpper(*it));
+    std::vector<MooseEnumItem>::iterator iter =
+        std::find_if(_current.begin(), _current.end(), [upper](MooseEnumItem const & item) {
```

Comment:
I think this should just stay as is, remove_if still needs to call erase and it is more obscure. If you really want me to change I can, but I don't think it is needed.